core/trino-main/src/main/java/io/trino/memory/NodeMemoryConfig.java

```diff
@@ -50,6 +55,24 @@ public NodeMemoryConfig setMaxQueryMemoryPerNode(DataSize maxQueryMemoryPerNode)
         return this;
     }
 
+    @NotNull
+    public Optional<DataSize> getMaxQueryMemoryPerTask()
+    {
+        return maxQueryMemoryPerTask;
+    }
+
+    @Config(QUERY_MAX_MEMORY_PER_TASK_CONFIG)
+    public NodeMemoryConfig setMaxQueryMemoryPerTask(DataSize maxQueryMemoryPerTask)
+    {
+        if (maxQueryMemoryPerTask.toBytes() == 0) {
```

Comment:
Actually `0` has special meaning: "no-limit". Added description.

Reply:
Should it be allowed to be empty instead?

Reply:
It can be. I somewhat prefer it to be possible to explicitly set the value to "unlimited" in the config. Via session variable we can make unlimited if value is set to `null` - but I  am not very convinced it is easier to comprehend that special `0`.

@findepi any opinion here?

Reply:
I remember generally we prefer not to have values with a special meaning. To avoid enforcing this limit not setting the limit seems to be more intuitive than setting it to 0.

Reply:
Why would anyone set the property to 0 if default is no-limit?

Reply:
Currently no. I was more thinking of case when (if) we decide that default is not no-limit. I can change it though.